transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -1364,6 +1374,65 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
         public void channelReadComplete(ChannelHandlerContext ctx) throws Exception { }
     }
 
+    ChannelInboundHandler tailDelegate() {
+        return tail instanceof WrappedTailContext ? ((WrappedTailContext) tail).delegate : null;
+    }
+
+    private static final class WrappedTailContext extends AbstractChannelHandlerContext
+                                                  implements ChannelInboundHandler {
+        private static final String TAIL_NAME = generateName0(WrappedTailContext.class);
+        final ChannelInboundHandler delegate;
+
+        WrappedTailContext(DefaultChannelPipeline pipeline, ChannelInboundHandler tail) {
+            super(pipeline, null, TAIL_NAME, true, false);
+            this.delegate = checkNotNull(tail, "tail");
+        }
+
+        @Override
+        public ChannelHandler handler() {
+            return this;
+        }
+
+        @Override
+        public void channelRegistered(ChannelHandlerContext ctx) throws Exception { }
+
+        @Override
+        public void channelUnregistered(ChannelHandlerContext ctx) throws Exception { }
+
+        @Override
+        public void channelActive(ChannelHandlerContext ctx) throws Exception { }
+
+        @Override
+        public void channelInactive(ChannelHandlerContext ctx) throws Exception { }
+
+        @Override
+        public void channelWritabilityChanged(ChannelHandlerContext ctx) throws Exception { }
+
+        @Override
+        public void handlerAdded(ChannelHandlerContext ctx) throws Exception { }
+
+        @Override
+        public void handlerRemoved(ChannelHandlerContext ctx) throws Exception { }
+
+        @Override
+        public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+            delegate.userEventTriggered(ctx, evt);
+        }
+
+        @Override
+        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+            delegate.exceptionCaught(ctx, cause);
+        }
+
+        @Override
+        public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+            delegate.channelRead(ctx, msg);
+        }
+
+        @Override
+        public void channelReadComplete(ChannelHandlerContext ctx) throws Exception { }
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1186)